server/models/User.js

```diff
@@ -0,0 +1,26 @@
+const mongoose = require('mongoose');
+
+const UserSchema = new mongoose.Schema({
+	name: {
+		type: String,
+		required: true
+	},
+	email: {
+		type: String,
+		required: true
+	},
+	password: {
+		type: String,
+		required: true
+	},
+	my_lists: {
+		type: mongoose.Schema.Types.ObjectId,
+		ref: 'Product List'
+	},
+	friends_list: {
+		type: mongoose.Schema.Types.ObjectId,
+		ref: 'Friends List'
+	}
```

Comment:
I guess a user may have more than one shopping list and more than one friend. If we use an array to handle these two data fields, would this schema better fit users' need?

Reply:
Hmmm Yeah that's what I was initially thinking, what do you think @surhud004 ?

Reply:
Maybe I didn't express my thought clearly. What I mean may look like this.
my_lists: [{
        type: mongoose.Schema.Types.ObjectId,
	ref: 'Product List'
    }],
friends_list: [{
	type: mongoose.Schema.Types.ObjectId,
	ref: 'Friends List'
	}]

Reply:
That is correct @ChianHuei. It is basically an array of Product schemas and similarly, an array of List schemas in User model.

Reply:
The ref in my_lists would be Product and the ref in friends_list would be User. It is the name of the schema which we defined.

Reply:
@surhud004 Excuse me. Could you please explain more? Do you mean we don't have to put `[ ]` on these two data fields or declare `type: array`, and these data fields will be an array by default?? But the type of your listSchema is not array.https://github.com/hatchways/team-watermelon/blob/feature-BE-lists-prods/server/models/list.js Also, I though the ref in my_lists was for shopping list IDs and we would use `user.populate(my_lists)` to know how many products in each list respectively. But, it looks like you have other ideas. Finally, I think if the my_lists in userSchema does not contain shopping list ID, then, your listSchema must include a reference for userID. Otherwise, there will be no connection between user ID and list ID, right ?  Thanks.